examples/widgets_minimal/src/main.rs

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! This example provides demostrations of building a Qt Widgets CXX-Qt application
+
+use cxx_qt_lib::QString;
+use cxx_qt_lib_extras::{QApplication, QPushButton};
+
+fn main() {
+    let mut app = QApplication::new();
+
+    // TODO: we should really pass a parent in here
+    let mut push_button = QPushButton::new();
```

Comment:
Should we use UniquePtr or should we use raw pointer and set a parent like Qt, or should we support both?